sympy/functions/elementary/trigonometric.py

```diff
@@ -2261,6 +2295,20 @@ def eval(cls, arg):
         if i_coeff is not None:
             return S.ImaginaryUnit * atanh(i_coeff)
 
+        if isinstance(arg, tan):
+            ang = arg.args[0]
+            if ang.is_comparable:
+                ang %= pi # restrict to [0,pi)
+                if ang > pi/2: # restrict to [-pi/2,pi/2]
+                    ang = pi - ang
```

Comment:
This line should be `ang = ang - pi`, as tan is pi-periodic. `tan(pi - ang)` is not tan(ang).   For example, 
```
>>> atan(tan(15))
-15 + 5*pi
```
is wrong. (And this example should be added to the tests).

Reply:
Yes, good catch. Thanks, I will add it.